substrate/client/tracing/src/logging/fast_local_time.rs

```diff
@@ -76,7 +87,8 @@ thread_local! {
 }
 
 impl FormatTime for FastLocalTime {
-	fn format_time(&self, w: &mut dyn Write) -> std::fmt::Result {
+	fn format_time(&self, w: &mut dyn Write) -> fmt::Result {
```

Comment:
Can you rewrite this so that:

1) There are two timestamp caches, one for local time and another for UTC. (This will make `CachedTimestamp::last_utc` unnecessary.)
2) The body of this function would essentially be something like this:

```
// Outside:
fn format_time_local(...) { ... }
fn format_time_utc(...) { ... }
// ...
// Inside::
let unix_time = ...;
if self.utc {
    format_time_utc(...)
} else {
    format_time_local(...)
}
```

This way there will only be a single toplevel `if self.utc` check.